### ISSUE TITLE
Perf improvement: make CQL driver use optimized `CqlVector<Float>` codec to improve performance

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/data-api#1775]
Since the default Java CQL driver codec for Vectors (one that uses `CqlVector`) has performance issues (both due to `List<Float>` wrapping and huge number of temporary `ByteBuffer`s created), we'd like to optimize this.

And as per @absurdfarce's note, there is actually alternate codec (`FloatVectorToArrayCodec` accessed via `ExtraTypeCodecs.floatVectorToArray(int)`) which seems like it'd solve the issue.
I am not yet 100% sure how to enable its use (mostly due to it needing vector dimension), but if we can use it that should lead to sizable performance improvement wrt driver usage, for Vector-enabled Collections and Tables.

**Comment by tatu-at-datastax** [comment]
I do not quite see a way to register this codec in a way that can be dynamically handled, without sub-classing `CodecRegistry`. Problem is that codec must be constructed for specific Vector length, but existing registries assume static set of codecs... and we cannot really pre-create codec for all possible Vector length.

**Comment by amorton** [comment]
ok, sounds like we can hold on this until the new year

**Comment by tatu-at-datastax** [comment]
@amorton Yes, will possible need help from driver team on this one; registration complicated by the need to probably construct different codec for different Vector columns (due `dimension` differing).

**Comment by tatu-at-datastax** [comment]
Brett has a PoC to use as baseline:

* Branch with changes: https://github.com/absurdfarce/cassandra-java-driver/tree/subtype-only-vector-codec
* Diff for things we could copy-paste: https://github.com/apache/cassandra-java-driver/compare/4.x...absurdfarce:cassandra-java-driver:subtype-only-vector-codec?expand=1

the idea being that these changes probably won't go in mainline, but also do not require changes to driver. The only part that didn't work with existing "extra" codec is that it can only register for specific vector length values; included codec registers for any length of Float vector and checks things dynamically.
